source/slang-capture-replay/slang-filesystem.cpp

```diff
@@ -27,12 +32,31 @@ namespace SlangCapture
         return nullptr;
     }
 
+    // TODO: There could be a potential issue that could not be able to dump the generated file content correctly.
+    // Details: https://github.com/shader-slang/slang/issues/4423.
     SLANG_NO_THROW SlangResult FileSystemCapture::loadFile(
                 char const*     path,
                 ISlangBlob** outBlob)
     {
         slangCaptureLog(LogLevel::Verbose, "%p: %s, :%s\n", m_actualFileSystem.get(), __PRETTY_FUNCTION__, path);
         SlangResult res = m_actualFileSystem->loadFile(path, outBlob);
+
+        // Since the loadFile method could be implemented by client, we can't guarantee the result is always as expected,
+        // we will check every thing to make sure we won't crash at writing file.
+        //
+        // We can only dump the file content after this 'loadFile' call, no matter this call crashes or file is not
+        // found, we can't save the file anyway, so we don't need to pay special care to the crash recovery. We will
+        // know something wrong with the loadFile call if we can't find the file in the capture directory.
+        if ((res == SLANG_OK) && (*outBlob != nullptr) && ((*outBlob)->getBufferSize() != 0))
```

Comment:
If the inner load file call failed, we should also record this and make sure we return the same error code during replay.